### PR TITLE
Increase code coverage one

### PR DESCRIPTION
**Pull request by TheRealHaui**
Added various improvements to selected Unit Tests.

### CODE REVIEW

File: src/test/java/org/apache/commons/imaging/ImageDumpTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Unit tests for class {@link ImageDump}.
+ *
+ * @date 2017-07-13
+ * @see ImageDump
+ *
+ **/
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/ImageDumpTest.java]: I question the presence of Javadoc in this class and all the other classes affected by this PR. AFAICT not all the tests (if any) even have Javadocs, so why should javadocs be added for this class and the others?

File: src/test/java/org/apache/commons/imaging/ImageDumpTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class ImageDumpTest{
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/ImageDumpTest.java]: The `{` seems to be joined directly with the class name to its left. This is inconsistent with all the other test classes AFAICT.

File: src/test/java/org/apache/commons/imaging/ImageDumpTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void testDumpColorSpace() {
+
+        ImageDump imageDump = new ImageDump();
+        ColorSpace colorSpace = ColorSpace.getInstance(1004);
+        imageDump.dumpColorSpace("Ku&]N>!4'C#Jzn+", colorSpace);
+
+        assertEquals(3, colorSpace.getNumComponents());
+
+    }
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/ImageDumpTest.java]: Most test methods I've seen in commons-imaging prefer to have no blank lines at the start and end of the test method body. I suggest removing the blank lines at the start and end of `testDumpColorSpace`'s body here for consistency, so like this:

```java
@Test 
public void testDumpColorSpace() {
ImageDump imageDump = new ImageDump();
        ColorSpace colorSpace = ColorSpace.getInstance(1004);
        imageDump.dumpColorSpace("Ku&]N>!4'C#Jzn+", colorSpace);

        assertEquals(3, colorSpace.getNumComponents());
}

File: src/test/java/org/apache/commons/imaging/ImageDumpTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void testDump() {
+
+        ImageDump imageDump = new ImageDump();
+        BufferedImage bufferedImage = new BufferedImage(10, 10, 10);
+        imageDump.dump(bufferedImage);
+
+        assertEquals(10, bufferedImage.getHeight());
+
+    }
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/ImageDumpTest.java]: Ditto regarding blank lines.

File: src/test/java/org/apache/commons/imaging/ImageDumpTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Unit tests for class {@link ImageDump}.
+ *
+ * @date 2017-07-13
+ * @see ImageDump
+ *
+ **/
+public class ImageDumpTest{
+
+
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/ImageDumpTest.java]: Having two blank lines here seems to be inconsistent with the other test classes in commons-imaging. I suggest shortening it to one blank line.

File: src/test/java/org/apache/commons/imaging/ImageDumpTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        ImageDump imageDump = new ImageDump();
+        BufferedImage bufferedImage = new BufferedImage(10, 10, 10);
+        imageDump.dump(bufferedImage);
+
+        assertEquals(10, bufferedImage.getHeight());
+
+    }
+
+
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/ImageDumpTest.java]: Having two blank lines here seems to be inconsistent with the other test classes in commons-imaging. I suggest shortening it to one or zero blank lines.

File: src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Unit tests for class {@link RgbBufferedImageFactory}.
+ *
+ * @date 2017-07-13
+ * @see RgbBufferedImageFactory
+ *
+ **/
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java]: Ditto regarding javadoc.

File: src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+public class RgbBufferedImageFactoryTest{
+
+
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java]: Ditto regarding blank lines. Please shorten them for all other classes as well.

File: src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    @Test
+    public void testGetColorBufferedImageThrowsIllegalArgumentException() {
+
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java]: Ditto regarding blank lines. Please shorten them for all other classes as well.

File: src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        RgbBufferedImageFactory rgbBufferedImageFactory = new RgbBufferedImageFactory();
+
+        try {
+            rgbBufferedImageFactory.getColorBufferedImage(0, 0, true);
+            fail("Expecting exception: IllegalArgumentException");
+        } catch(IllegalArgumentException e) {
+            assertEquals("Width (0) and height (0) cannot be <= 0",e.getMessage());
+            assertEquals(DirectColorModel.class.getName(), e.getStackTrace()[0].getClassName());
+        }
+
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java]: Ditto regarding blank lines. Please shorten them for all other classes as well.

File: src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/RgbBufferedImageFactoryTest.java]: Ditto regarding blank lines. Please shorten them for all other classes as well.

File: src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java [hunk 12]
```diff
@@ -17,24 +17,22 @@
...
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.OutputStream;
+import java.io.*;
 
-import org.apache.commons.imaging.ImagingTest;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java]: I'd keep the ordering and style of the imports as before; no need to rearrange them just to import `assertEquals` and `fail`.

File: src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java [hunk 13]
```diff
@@ -17,24 +17,22 @@
...
 
 public abstract class ByteSourceTest extends ImagingTest {
     protected File createTempFile(final byte src[]) throws IOException {
         final File file = createTempFile("raw_", ".bin");
 
         // write test bytes to file.
         try (FileOutputStream fos = new FileOutputStream(file);
-                OutputStream os = new BufferedOutputStream(fos)) {
+             OutputStream os = new BufferedOutputStream(fos)) {
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java]: No need to un-tab this line AFAICT.

File: src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java [hunk 14]
```diff
@@ -66,6 +64,22 @@ protected File createTempFile(final byte src[]) throws IOException {
         }
         final byte longArray[] = (baos.toByteArray());
 
-        return new byte[][] { emptyArray, single, simple, zeroes, longArray, };
+        return new byte[][]{emptyArray, single, simple, zeroes, longArray,};
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java]: No need to remove spaces here AFAICT.

File: src/test/java/org/apache/commons/imaging/formats/bmp/BmpWriterRgbTest.java [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    public void testGetImageData() {
+
+        BmpWriterRgb bmpWriterRgb = new BmpWriterRgb();
+        BufferedImage bufferedImage = new BufferedImage(2, 2, 5);
+        byte[] byteArray = bmpWriterRgb.getImageData(bufferedImage);
+
+        assertEquals( 24, bmpWriterRgb.getBitsPerPixel() );
+        assertEquals( 0, bmpWriterRgb.getPaletteSize() );
+        assertEquals( 16, byteArray.length );
+        assertEquals( "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]", Arrays.toString(byteArray) );
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/formats/bmp/BmpWriterRgbTest.java]: Spaces at start and end of parameter lists seem unnecessary here.

File: src/test/java/org/apache/commons/imaging/formats/jpeg/iptc/IptcTypeLookupTest.java [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+public class IptcTypeLookupTest{
+
+
+    @Test
+    public void testGetIptcTypeWithPositive() {
+
+        IptcType iptcType = IptcTypeLookup.getIptcType(1117);
+
+        assertEquals( 1117, iptcType.getType() );
+        assertEquals( "Unknown", iptcType.getName() );
```
Comment (jbduncan) [inline src/test/java/org/apache/commons/imaging/formats/jpeg/iptc/IptcTypeLookupTest.java]: Ditto regarding spaces at start and end of parameter lists.

**Comment by garydgregory** [comment]
You want a fresh PR, yes, but you still need to base it off of git master, not just continue the previous patch please.

**Comment by TheRealHaui** [comment]
Done in #29.

**Pull request closed without merging**